### ISSUE TITLE
Cannot read property 'destroy' of undefined

### ISSUE THREAD

**Issue by sradevski** [whawker/react-jsx-highcharts#14]
Good job on wrapping HighCharts, it is a pain to work with it otherwise!

I have version 1.3.1, and I get an error even with the simplest example in the demos. It seems that somewhere the chart reference is deleted before the BaseChart component's componentWillUnmount event is called, thus the error. 

You can investigate why this occurs, but for now a simple solution is to check the existence of the chart before calling destroy().

So **BaseChart.js** componentWillUnmount should be 
```
componentWillUnmount () {
   
 if(this.chart){
       this.chart.destroy();
  }
  this.chart.__destroyed = true;
}
```

This might result in some memory leakage, but it might be a good quick fix for now.

**Comment by whawker** [comment]
Hmm I'm not sure why the reference would be deleted. I've never seen that before. 

What version of React are you using?

**Comment by sradevski** [comment]
I am not at work currently so I can't check, but since I upgraded our dependencies just few days ago it should be 15.6.1.

**Comment by whawker** [comment]
Ok thanks, by any chance is the component mounted, then unmounted almost immediately? I can't think of many reasons why `this.chart` would be undefined.

**Comment by sradevski** [comment]
I didn't really spend a lot of time debugging, I just added an additional check so I can continue with development, but I can have a look if it helps.

**Comment by sradevski** [comment]
I found the cause of it. I have wrapped the chart in another component, that only renders its children when the status is not loading. It seems React instantiates the children before they are rendered, and then immediately unmounts them, before the this.chart object is created.

```
<StatusWrapper
          isLoading={isAnalysisDataFetching}
          errorMessage={analysisDataError}
          <HighGmsChart height={450} seriesData={seriesData} />
</StatusWrapper>
```

Render method of StatusWrapper:
```
...
<div style={style}>
      {Spinner || ErrorMessage || children}
</div>
```

**Comment by whawker** [comment]
Interesting, I didn't know about that React behaviour. 

Yep I'll wrap the `componentDidUnmount` in an if statement and push out a new release.

**Comment by whawker** [comment]
Published as 1.3.2, thanks for the heads up and debugging.
